### ISSUE TITLE
Refresh Project Estimate component on change of number of screens

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/connect-app#691]
### Expected behavior
Project estimate should refresh itself as soon as number of screens value is changed by user.

### Actual behavior
Right now it is refreshed only after clicking Save button

### Steps to reproduce the problem
- Change number of screens value in a project which is already in Review stage
- Notice that project estimate component changes/refreshes itself only when Save button in clicked.

--

#### Environment
- OS: All
- Browser (w/version): All
- User role (client, copilot or manager): Any role who has access to a project.
- Account used: n/a

**Comment by vikasrohit** [comment]
@fnisen @vic-topcoder please let us know if you think differently for this use case?

**Comment by vikasrohit** [comment]
@gondzo if we can't get it fixed before bug hunt, we can put it in known issue list.

**Comment by fnisen** [comment]
@vikasrohit yes, agree. It's supposed to be a "quick quote"--meaning, you select an option and see the price right away. Related ticket: https://github.com/appirio-tech/connect-app/issues/677

**Comment by vikasrohit** [comment]
@gondzo Can you please grab this one?

**Comment by gondzo** [comment]
Sure, working on it..

**Comment by fnisen** [comment]
@gondzo can you please check that whatever is done here doesn't overlap with @vikasrohit's work on https://github.com/appirio-tech/connect-app/issues/691, which already has an open PR

**Comment by vikasrohit** [comment]
@fnisen thanks for actively alerting Samir. However, I don't think I have any work related to this ticket in pending. So, no need to worry.

**Comment by gondzo** [comment]
@all Prize increased

**Comment by iit2009060** [comment]
@gondozo Can you please describe the behaviour , Currently I added the screen in the specification tab and then refresh the page I am seeing the same project estimate(Duration)  value.Can you give example or screenshot which can help me to solve this issue faster.

**Comment by gondzo** [comment]
@iit2009060 it's not about refreshing the page. As soon as you select a different option in the number of screens section, the project estimate section should be updated (even without clicking save button). Does that help ?

**Comment by gondzo** [comment]
project estimate should change without saving the project >
![image](https://cloud.githubusercontent.com/assets/3051646/23899508/adf8cb96-08b5-11e7-8688-ef06b4834889.png)
![image](https://cloud.githubusercontent.com/assets/3051646/23899517/b8d99950-08b5-11e7-8e9d-13abd434868b.png)

**Comment by iit2009060** [comment]
@gondozo can you give the project id where I can make changes to the screen number selection.I tried some other projects but they are not allowing me to change it and not displaying the error.

**Comment by gondzo** [comment]
Just create a new project in design/application visual design

**Comment by iit2009060** [comment]
@gondzo  I saw some previous issue like this https://github.com/appirio-tech/connect-app/issues/727 where if description is empty it still create the error that project name is empty ,this issue does not handle the description  is empty.If you can create a new issue I can create a PR for this in half an hour.This issue require more understanding of code which i am doing it  currently.

**Comment by gondzo** [comment]
@created #835 and assigned to you. #691 is still a higher priority, so finish it before moving on to 835

**Comment by iit2009060** [comment]
@gondzo  these are the sibling component of SpecContainer .I am not getting how we can make sibling components interact ,do you have any suggestion ?

**Comment by iit2009060** [comment]
@gondzo  it is taking lots of time , I  tried changing it but breaking other things . If I will have a working solution i will pick it up again.

**Comment by vikasrohit** [comment]
@iit2009060 @gondzo Can we please hold this for some time? I think I already have a solution for this as by product of fixing another issue.

**Comment by vikasrohit** [comment]
@gondzo Can we safely remove it from Open from Pickup?

**Comment by gondzo** [comment]
yes

**Comment by fnisen** [comment]
@vikasrohit I'm not seeing this work in Dev after your update.

**Comment by vikasrohit** [comment]
Sorry @fnisen Actually build is failing for some unknown reason. As soon as build works out, changes should be in dev. @parthshah Can you please help me here?

**Comment by vikasrohit** [comment]
Now deployed into dev. Verified. It was some dependency conflicting with js uglify plugin for webpack.

**Comment by fnisen** [comment]
@vikasrohit I'm seeing an edge case where this doesn't work. Please try this: 
- Create AVD Concepts Projects
- Select a screen size, fill out the first section, and Save
- Click on screen size 4-8 or 9-15. Price adjusts. Click on size 1-3. Price does not adjust. 
- Now make a change to the form. Price for 1-3 now adjusts.
Here is a video: https://www.screencast.com/t/xeoYR3qxXCX

**Comment by vikasrohit** [comment]
Thanks @fnisen for verifying the issue and finding the edge case. It was really an edge case. Actually, I have put a check to optimize the performance, but it seems that is preventing the components (outside of the form) to update when user changes/reverts answer such that it matches to the saved values of that question.
It should be fixed now in dev. Please let me know if you still see the same issue or anything else around this.

**Comment by fnisen** [comment]
@vikasrohit working in Dev. Thank you! 👍
